crates/core/src/client/waiter.rs

```diff
@@ -0,0 +1,97 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use starknet::core::types::{MaybePendingTransactionReceipt, StarknetError, TransactionReceipt, TransactionStatus};
+use starknet::providers::{MaybeUnknownErrorCode, Provider, ProviderError, StarknetErrorWithMessage};
+use starknet_crypto::FieldElement;
+use tokio::time::Instant;
+
+use super::errors::EthApiError;
+
+/// A helper struct for waiting for a transaction to be mined.
+/// Inspired by https://github.com/dojoengine/dojo/blob/main/crates/dojo-world/src/utils.rs
+pub struct TransactionWaiter<P: Provider> {
+    provider: Arc<P>,
+    transaction_hash: FieldElement,
+    interval: Duration,
+    timeout: Duration,
+}
+
+impl<P: Provider> TransactionWaiter<P> {
+    pub fn new(
+        provider: Arc<P>,
+        transaction_hash: FieldElement,
+        interval_in_milli: u64,
```

Comment:
```
ms_interval: u64,
```

or 
```
interval_ms
```

no?

Reply:
I was trying to stay consistent with crates such as `std::time` which use millis. I could update to `interval_millis`